app/helpers/recent_items_helper.rb

```diff
@@ -0,0 +1,49 @@
+# A random selection of recently modified works to display as thumbnails
+# on the front page of the site.
+# The same selection is shown to all users at the same time.
+# The selection is reshuffled few minutes,
+# and is cached for speed.
+module RecentItemsHelper
+  class RecentItems
+    @@last_refresh = nil
+    @@bag = nil
+
+    def initialize()
+      @how_many_works_to_show = 6
+      #@how_often_to_change = 60 * 10 # ten minutes
+      @how_often_to_change = 5
+      @how_many_works_in_bag = 15
+    end
+
+    def recent_items()
+      @@bag = nil if time_for_a_new_bag?
+      srand @@last_refresh
```

Comment:
I'm sorry I'm just noticing this, but I think srand will seed the ruby pseudo-random number generator _program wide_, you've changed how random numbers are generated everywhere. 

Which also might not result in the repeatability you want, if other parts of the program use the random number generator in between your calls to it, which they very well might. 

I wonder if there's a way to do this with just local effects.

Reply:
Looking at ruby docs, I think maybe we can do this:

```
generator = Random.new(seed)
generator.rand
```

To use a local one with our custom seed, instead of insisting on our custom seed program-wide. 

I guess this `generator` would _also_ have to be stored in a class variable, so repeated calls to it are to the same generator and get the sequence. 

I think this all shows how doing "clever" things incures a cost, more possibility for bugs and more time spent making it robust and reliable. I wonder if we want to do this clever rand thing at all, but I think let's stick with it for now to hopefully finally get done with this feature!